src/main/java/com/axis/jenkins/plugins/eiffel/eiffelbroadcaster/EiffelActivityTriggerAction.java

```diff
@@ -0,0 +1,77 @@
+/**
+ The MIT License
+
+ Copyright 2021 Axis Communications AB.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ */
+
+package com.axis.jenkins.plugins.eiffel.eiffelbroadcaster;
+
+import com.axis.jenkins.plugins.eiffel.eiffelbroadcaster.eiffel.EiffelActivityTriggeredEvent;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import hudson.model.Action;
+import javax.annotation.CheckForNull;
+import org.kohsuke.stapler.export.Exported;
+import org.kohsuke.stapler.export.ExportedBean;
+
+/**
+ * An {@link Action} for storing the
+ * <a href="https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelActivityTriggeredEvent.md">
+ * EiffelActivityTriggeredEvent</a> that was emitted when a build was enqueued.
+ */
+@ExportedBean
+public class EiffelActivityTriggerAction implements Action {
+    private String eventJSON;
```

Comment:
Is there a reason you don't store the Event directly? This will make it way harder to xtream migrations when the EiffelActivityTriggeredEvent changes.

Reply:
I actually thought it would be less of a hassle to store the wire representation of the event. If the event changes we may have to create a new event class (depending on what changes are made) and/or use some kind of custom Jackson deserializer, probably very similar to the custom XStream stuff we'd have to write if we store the event itself.

If you think XStream serialization is better I can certainly change. However, I still want the event JSON to be accessible via REST, but that could be solved with a custom endpoint. That's actually more elegant than having clients request $BUILD_URL/api/json and parsing the list of actions. Do you know of an example of something similar that I can steal^Wget inspiration from?

Reply:
I just figured that managing the conversions in the public getEvent() would be much more cumbersome than managing the conversion on startup. But I don't really have strong feelings about it.

Reply:
I'll go with the current implementation for now. Major protocol changes will probably be a PITA to manage either way. I've filed #24 to add a REST endpoint to obtain the ActT event which should make it possible for us to change the XStream representation without too much hassle.